packages/flutter_tools/test/flutter_manifest_test.dart

```diff
@@ -359,6 +359,80 @@ flutter:
       final FlutterManifest flutterManifest = await FlutterManifest.createFromString(manifest);
       expect(flutterManifest.isEmpty, false);
     });
+
+    test('test parsing appVersion major.minor.patch+build', () async {
+      const String manifest = '''
+name: test
+version: 1.0.0+2
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+''';
+      final FlutterManifest flutterManifest = await FlutterManifest.createFromString(manifest);
+      expect(flutterManifest.appVersion, '1.0.0+2');
+      expect(flutterManifest.buildName, '1.0.0');
+      expect(flutterManifest.buildNumber, 2);
+    });
+
+    test('test parsing appVersion major.minor+build', () async {
+      const String manifest = '''
+name: test
+version: 1.0+2
+dependencies:
+  flutter:
+    sdk: flutter
+flutter:
+''';
+      final FlutterManifest flutterManifest = await FlutterManifest.createFromString(manifest);
+      expect(flutterManifest.appVersion, '1.0+2');
+      expect(flutterManifest.buildName, '1.0');
+      expect(flutterManifest.buildNumber, 2);
+    });
```

Comment:
Duplication could be removed by creating a function like
```dart
checkManifestVersion({
  String versionLine,
  String expectedAppVersion,
  String expectedBuildName,
  int expectedBuildNumber,
}) { ...}
```
Specifying an empty `versionLine` covers the case where there is no `version:` clause in the `pubspec.yaml` file.

Reply:
done